public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts

```diff
@@ -66,10 +66,13 @@ export const useAdvancedMode = (
   );
 
   useEffect(() => {
-    if (!isAdvancedMode && isGrafanaAlertingType) {
-      setSimpleCondition(getSimpleConditionFromExpressions(expressionQueries));
+    if (isGrafanaAlertingType) {
+      setValue('editorSettings.simplifiedQueryEditor', !isAdvancedMode);
```

Comment:
This code is being called every time we edit anything in expression queries. It shouldn't be a big issue but I'm a bit concerned about potential side effects of that. 
I guess we want to change this value only if the user actually switches the toggle